types/got/got-tests.ts

```diff
@@ -266,6 +266,14 @@ got('/examples/angularjs', { baseUrl: 'http://todomvc.com' });
 got('http://todomvc.com', { headers: { foo: 'bar'} });
 got('http://todomvc.com', { cookieJar: new tough.CookieJar() });
 
+// Test extension
+got.extend({ method: 'POST' })('/example');
+got.extend({ method: 'POST' }).extend({ headers: {} }).stream('/example');
+
+got.extend({ json: true })('/example').then(({ body }) => body.value); // Body is `any`
```

Comment:
Comment `// Body is any` doesn't check anything. For check type you can:

1. Assign to another variable with type `const value: any = body.value`
2. Use `$ExpectType` from dtslint: https://github.com/DefinitelyTyped/DefinitelyTyped#lint`